drivers/modem/ublox-sara-r4.c

```diff
@@ -1252,6 +1252,12 @@ static ssize_t offload_write(void *obj, const void *buffer, size_t count)
 	return offload_sendto(obj, buffer, count, 0, NULL, 0);
 }
 
+static ssize_t offload_sendmsg(void *obj, const struct msghdr *msg, int flags)
+{
+	errno = -ENOTSUP;
```

Comment:
I think the errno value should be >0, just just do `errno = ENOTSUP;` here.

Reply:
Ah, of course - thanks for the note.

As a matter of fact I will make an attempt to implement sendmsg after all, following the discussion in #24519. So this code here is going to be replaced soon.

Reply:
I have now implemented sendmsg, and included in this PR.
This is tested for TCP using the MQTT library, but no tests are done for UDP packets.

Reply:
> I think the errno value should be >0, just just do `errno = ENOTSUP;` here.

Then there may be a similar problem in the simplelink wifi driver, where I borrowed these lines:
https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/wifi/simplelink/simplelink_sockets.c#L963

Reply:
> Then there may be a similar problem in the simplelink wifi driver, where I borrowed these lines:
> https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/wifi/simplelink/simplelink_sockets.c#L966

Yep, that is a bug in simplelink driver.